src/language-js/print/binaryish.js

```diff
@@ -76,7 +76,7 @@ function printBinaryishExpression(path, options, print) {
     (isCallExpression(parent) && parent.callee === node) ||
     parent.type === "UnaryExpression" ||
     (isMemberExpression(parent) && !parent.computed) ||
-    parent.type === "LogicalExpression" && needsParens(path, options)
+    needsParens(path, options)
```

Comment:
Why are you removing the parent type check? `??` is also `LogicalExpression` in estree.

Reply:
It seems that with the `parent.type` check prevents a logical expression from being indented if parent type is a binary expression (`parent.type === "BinaryExpression"`)

Input:
```
num_x *
  (
    long_long_long_long_long_long_long_condition_2 ??
      long_long_long_long_long_long_long_condition_3
  );
```

Output
```
num_x *
  (long_long_long_long_long_long_long_condition_2 &&
    long_long_long_long_long_long_long_condition_3);
```
If we remove the parent type check, then the output will be correct and match the expected output.

Expected Output
```
num_x *
  (
    long_long_long_long_long_long_long_condition_2 ??
      long_long_long_long_long_long_long_condition_3
  );
```

If you want I can add the parent type check back in.

Reply:
Let's add it back, only check `LogicalExpression` first.

Reply:
Ok, I've added the `LogicalExpression` `parent.type` check back.